examples/bzlmod/MODULE.bazel

```diff
@@ -0,0 +1,27 @@
+module(name = "example")
+
+bazel_dep(name = "buildbuddy_toolchain", repo_name = "buildbuddy")
+local_path_override(
+    module_name = "buildbuddy_toolchain",
+    path = "../../",
+)
+
+buildbuddy = use_extension("@buildbuddy//:extensions.bzl", "buildbuddy")
+
+# Example customizations:
+#
+# buildbuddy.platform(container_image = "docker://my-container.image/replace/me@sha256:1234567890abcdef")
+# buildbuddy.gcc_toolchain(
+#     gcc_major_version = "11",
+#     extra_cxx_builtin_include_directories = ["/some/include/dir"],
+# )
+# buildbuddy.msvc_toolchain(
+#     msvc_edition = "Enterprise",
+#     msvc_release = "2022",
+#     msvc_version = "14.39.33519",
+#     windows_kits_release = "10",
+#     windows_kits_version = "10.0.22621.0",
+# )
+
+# All relevant toolchain and platform targets should be available under "@buildbuddy_toolchain//..." repository
+use_repo(buildbuddy, "buildbuddy_toolchain")
```

Comment:
We can start with this for maximum flexibility, but we should consider allowing users to add custom constraints to the platforms and at the same time registering them for them.

Reply:
Agree. I think we can automatically register toolchains and exec platform in a future change. 

I think we do want to sort out the known issue with Go toolchain registration(which we discussed yesterday) first though. Or perhaps we can provide an opt-out config knob for folks to disable the behavior.